### PR TITLE
chore: handle error 404 and 500 in the charge_presentation

### PR DESCRIPTION
**Pull request by YogiPristiawan**
for this issue https://github.com/TokoBapak/mock-payment-provider/issues/4

### CODE REVIEW

**Comment by aldy505** [comment]
description PR nya harusnya `Closes #4`, biar langsung ke linked ke issue itu

File: presentation/cancel_transaction.go [hunk 1]
```diff
@@ -36,9 +36,39 @@ func (p *Presenter) CancelTransaction(w http.ResponseWriter, r *http.Request) {
 	if err != nil {
 		if errors.Is(err, business.ErrTransactionNotFound) {
 			// TODO: handle error, return 404
```
Comment (aldy505) [inline presentation/cancel_transaction.go:38]: hapus TODO nya

File: presentation/cancel_transaction.go [hunk 2]
```diff
@@ -36,9 +36,39 @@ func (p *Presenter) CancelTransaction(w http.ResponseWriter, r *http.Request) {
...
+			responseBody, e := json.Marshal(schema.Error{
+				StatusCode:    http.StatusBadRequest,
```
Comment (aldy505) [inline presentation/cancel_transaction.go:40]: StatusCode BadRequest kok kirim StatusNotFound di line 49?

**Review by aldy505 — CHANGES_REQUESTED** [review]

File: presentation/cancel_transaction.go (hunk 2 above)
Comment (YogiPristiawan) [reply presentation/cancel_transaction.go:40]: wkwk iya salah

**Review by aldy505 — APPROVED** [review]

**Pull request merged**